### PR TITLE
Holodeck Fixes

### LINKED ISSUE #13183: EMHolo machines can be disassembled
Emergency Medical machines (Sleeper, Chem master) can be disassembled into real non-holographic parts. These can then be researched or reassembled elsewhere at no cost.

### LINKED ISSUE #13277: You can create laser beams that are frozen in time using an emagged holodeck
Repro:
1. Emag the Holodeck console
2. Select Holdout Bunker
3. Turn the simulation off
4. GOTO 2
They're stackable and are actually capable of dealing damage

### PR DESCRIPTION
**Pull request by SuperSayu**
Adds NODECONSTRUCT flag to structures and machines.  This is specifically used by the holodeck for things that should be immune to disassembly.  Fixes #13183.
Fixes #13277 laser beams spawning when the holdout program ends

Fixes invisible windoors in the holodeck.
Standardizes the fire axe's interaction with windows.
Fixes a runtime when holodeck tiles get broken or burned.

### CODE REVIEW

**Comment by tkdrg** [comment]
:+1: it will be good to have a systems in place for this kind of thing

File: code/game/objects/items/weapons/twohanded.dm [hunk 1]
```diff
@@ -177,17 +177,12 @@
...
-			new /obj/item/weapon/shard( W.loc )
-			if(W.reinf) new /obj/item/stack/rods( W.loc)
-
-			if (W.dir == SOUTHWEST)
-				new /obj/item/weapon/shard( W.loc )
-				if(W.reinf) new /obj/item/stack/rods( W.loc)
-		qdel(A)
+			W.spawnfragments() // this will qdel and spawn shards
+		else if(istype(A,/obj/structure/grille))
+			qdel(A)
```
Comment (Anonus) [inline code/game/objects/items/weapons/twohanded.dm]: hit the grille for like 90 or something, so it spawns rods

**Comment by Aranclanos** [comment]
so... all of these checks just for the holodeck?

**Comment by SuperSayu** [comment]
They can also be used on away missions or on space derelict stuff that
should not be moved to the station.

The holodeck has been part of tgcode for a long time.  With the upgrades I
added it is more feasible to add holodeck programs with real stuff in
them.  If you feel this code is going to waste, add more interesting
holodeck stuff, or extra holodecks that do different things.  I could give
you examples off the top of my head--I did in my holodeck PR already.

Also if you can think of a more reasonable way I am happy to oblige.  These
checks only happen in specific attackbys, they won't slow anything down.

**Comment by Aranclanos** [comment]
I would probably just straight make new types for the holodeck instead of this.
like some esword rack that only accepts eswords to be stored, instead of a table that needs continuous fixing over the years

**Comment by SuperSayu** [comment]
I disagree largely because that puts the onus on mappers to code variants
of anything they want to use.  Figuring out what needs to change when
something is holographic or map-special means diving pretty hard into the
code for each object mapped.  This actively discourages mappers from making
anything with machines in it, and that's not a good thing.

The machinery deconstruct abstract checks cut out a lot of that but not
everything. It does provide a framework of how those changes could our
should be done when they are added to something new.  And using the same
flag across structures and machines is good, although I half-want to go
back and make it is own flag name that just overlaps with ABSTRACT
numerically so the flag if clearer.  If rather not though.

**Comment by Cheridan** [comment]
aran that's a really horrible way to go about it (and the way it used to be done)

basically if you wanted to have a table in the holodeck, you had to make a snowflake holotable without deconstruction code; repeat for, like, any structure you wanted to use ever.

that said i don't know if i like the ABSTRACT flag for this, since its original purpose was to simulate abstract types from other languages, items that weren't ever meant to be spawned in-game.

**Comment by SuperSayu** [comment]
Alright well since I have to fix conflicts again I will change the flag name and probably do minimal Z2 mapfixen

Razzem frazzem slow merge

**Pull request merged**